jupyter_ui_poll/_poll.py

```diff
@@ -52,7 +52,9 @@ def __init__(self, shell, loop) -> None:
             # ipykernel < 6
             kernel.shell_handlers["execute_request"] = self._execute_request
 
-        shell.events.register("post_execute", self._post_execute_hook)
+        # previously, we used "post_execute", but a comm handler can also trigger this event,
+        # Instead, if we assume silent=False, we can better use "post_run_cell"
```

Comment:
Can you explain more about `silent=False` part?

Reply:
Yes, on https://github.com/ipython/ipykernel/blob/95f245138589db4f723b8af06107cdfeadc53314/ipykernel/ipkernel.py#L451 we see this gets skipped.

I am not sure when silent=True happens to be honoust, I guess when something gets executed that is not a cell execution?
Looking at https://github.com/ipython/ipykernel/blob/95f245138589db4f723b8af06107cdfeadc53314/ipykernel/kernelbase.py#L786 Maybe a good way to solve this is not just directly execute code when silent=True, althoug that might lead to strange behaviour.